__tests__/dedent-tests.js

```diff
@@ -27,6 +27,29 @@ describe("dedent", () => {
     ).toMatchSnapshot();
   });
 
+  it("works with multiline interpolation", () => {
+    expect(
+      dd`
+        first
+          ${"second\nthird"}
+        fourth
+      `
+    ).toMatchSnapshot();
+  });
+
+  it("works with dedent interpolation", () => {
+    expect(
+      dd`
+        first
+          ${dd`
+            second
+            third
+          `}
+        fourth
+      `
+    ).toMatchSnapshot();
+  });
```

Comment:
Probably worth adding a test like the one referenced in #12 , namely:

```jsx
const weird = dedent`
  foo.
  ${things.map(x => dedent`
    hello
      world
  `).join('\n')}
  bar.
`;
```
which should produce (as it does): 
```
foo.
hello
  world
hello
  world
hello
  world
bar.
```
Possibly also 
```jsx
const weirder = dedent`
  foo.
    ${things.map(x => dedent`
      hello
        world
    `).join('\n')}
  bar.
`;
```
which I think should produce
```
foo.
  hello
    world
  hello
    world
  hello
    world
bar.
```

Reply:
I feel these extra tests are not necessary, as this is covered by the 'non-map' version of the tests... dunno how the author feels about tests testing the same thing :P

Reply:
... Yes, you're correct; I wasn't reading very carefully. Thanks!

The inner template should probably include indents/dedents, though, eg; 

```jsx
dd`
  first
    ${dd`
         second
       third
           fourth
    `}
  fifth
`
```

to ensure that "inner template" logic is the same as "outer template" logic

Reply:
(to be clear, I think the above is purely optional and I would expect it to pass today)